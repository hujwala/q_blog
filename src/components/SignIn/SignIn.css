
.image { 
	width: 500px;
	height: 500px;
}

.title {
	font-size: 50px;
	text-align: center;
}
.form-group {
 	width: 60%;
 	margin-left: 20%;
}

.button-submit{
	margin-right: 120px;
    margin-left: 20px;
}
.mt80{
	margin-top: 80px;
}

.singin-button{
	background: #e6932b;
    color: white;
}

.singin-button:hover{
	background: #e6932b;
    color: white;
}
