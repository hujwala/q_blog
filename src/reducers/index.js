import { combineReducers } from 'redux';
import api_reducer from './api_reducer/api_reducer';
import profile_reducer from './profile_reducer/profile_reducer'
import { reducer as formReducer } from 'redux-form';

export default combineReducers({
  form: formReducer, 	
  api: api_reducer,
  profileDetails: profile_reducer
})