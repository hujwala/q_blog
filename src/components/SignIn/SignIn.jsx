import React from 'react';
import {Component} from 'react';
import { Field, reduxForm } from 'redux-form';
import './SignIn.css'
import { bake_cookie, read_cookie, delete_cookie } from "sfcookies";



const renderField = ({ input, label, type, meta: { touched, error, warning } }) => (
    <div>
      <label className="control-label">{label}</label>
      <div>
        <input {...input} placeholder={label} type={type} className="form-control" />
        {touched && ((error && <span className="text-danger">{error}</span>) || (warning && <span>{warning}</span>))}
      </div>
    </div>
  )

const validate = values => {
    const errors = {}
    if (!values.email) {
      errors.email = 'Required'
    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
      errors.email = 'Invalid email address'
    }
    if (!values.password) {
        errors.password = 'Required'
      } else if (values.password.length < 2) {
        errors.password = 'Minimum be 2 characters or more'
      }
    return errors
  }

class SignIn extends Component {
  state = {
    toLogin: false
  }

  submit = (values) => {
    const user = {
      email: values.email,
      password: values.password
    };
    fetch("http://192.168.2.115:8080/getUser", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        'Access-Control-Allow-Origin':'*'
      },
      body: JSON.stringify(user)
    })
    .then(response => {
        if (response.status === 200) {
          bake_cookie("auth_token", response.headers.get("Authorisation"));
        }
        return response.json();
      });
  }

render(){
  let { fields: {email, password}, handleSubmit, pristine, submitting } = this.props;
    return (
    <div className="container">
    <div className="row mt80">
    <div className="col-md-6 col-sm-6 col-lg-6">
    <img src={require("../../image/blog.jpg")} alt="boohoo" className="img-responsive" className="image"/>
    </div>
    <div className="col-md-6 col-sm-6 col-lg-6">
    <form onSubmit={ handleSubmit(props => this.submit(props))} >
      <div className='title'>Sign In </div>
      <div className="form-group">
        <Field name="email" component={renderField} label="Email" {...email} className="form-control"/>
      </div>
      <div className="form-group">
        <Field name="password" component={renderField} label="Password" type="password" {...password} className="form-control"/>
      </div>
      <div className="form-group">
        <button type="submit" className="btn singin-button">Submit</button>
      </div>
    </form>
    </div>
    </div>
    </div>
  )
  }
}

export default reduxForm({
  form: 'contact',
  fields: ['email', 'password'],
  validate
})(SignIn);
