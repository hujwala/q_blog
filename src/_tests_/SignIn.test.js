import React from 'react';
import { shallow, mount, render, configure } from 'enzyme';
import SignIn, { renderTextInput } from '../components/SignIn/SignIn';
import ReduxThunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import reducers from '../reducers';
import Adapter from 'enzyme-adapter-react-16';
import { reducer as formReducer, SubmissionError } from 'redux-form';
import { createStore, combineReducers } from 'redux'
import { Provider } from 'react-redux'

configure({ adapter: new Adapter() });

describe("SignIn", () => {
  let store
  let submit
  let subject
  beforeEach(() => {
    store = createStore(combineReducers({ form: formReducer }))
    submit = sinon.stub().returns(Promise.resolve())
    const props = {
      submit,
    }
    
    subject = mount(
      <Provider store={store}>
        <SignIn {...props}/>
      </Provider>
    )
  })
  it('should have form', () => {
    // expect(subject.find('form').hasClass('login')).toBe(true)
    expect(subject.find('Field').length).toEqual(2);
  })

  it('calls on submit', ()=> {
        const form = subject.find('form');
        subject.find('#email').simulate('change', {target: {name: 'email', value: 'sunil.kr1024@gmail.com'}, preventDefault: () => {}})
        subject.find('#password').simulate('change', {target: {name: 'password', value: 'qwinix123!'}, preventDefault: () => {}})
        form.simulate('submit');
        expect(toJson(form)).toMatchSnapshot();
    });
});