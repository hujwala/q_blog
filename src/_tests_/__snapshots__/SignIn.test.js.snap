// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn calls on submit 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}
>
  <div
    className="title"
  >
    Sign In 
  </div>
  <div
    className="form-group"
    id="email"
  >
    <Field
      className="form-control"
      component={[Function]}
      label="Email"
      name="email"
    >
      <Connect(ConnectedField)
        _reduxForm={
          Object {
            "anyTouched": false,
            "array": Object {
              "insert": [Function],
              "move": [Function],
              "pop": [Function],
              "push": [Function],
              "remove": [Function],
              "removeAll": [Function],
              "shift": [Function],
              "splice": [Function],
              "swap": [Function],
              "unshift": [Function],
            },
            "arrayInsert": [Function],
            "arrayMove": [Function],
            "arrayPop": [Function],
            "arrayPush": [Function],
            "arrayRemove": [Function],
            "arrayRemoveAll": [Function],
            "arrayShift": [Function],
            "arraySplice": [Function],
            "arraySwap": [Function],
            "arrayUnshift": [Function],
            "asyncErrors": undefined,
            "asyncValidate": [Function],
            "asyncValidating": false,
            "autofill": [Function],
            "blur": [Function],
            "change": [Function],
            "clearAsyncError": [Function],
            "clearFields": [Function],
            "clearSubmit": [Function],
            "clearSubmitErrors": [Function],
            "destroy": [Function],
            "destroyOnUnmount": true,
            "dirty": false,
            "dispatch": [Function],
            "enableReinitialize": false,
            "error": undefined,
            "fields": Array [
              "email",
              "password",
            ],
            "focus": [Function],
            "forceUnregisterOnUnmount": false,
            "form": "contact",
            "getFormState": [Function],
            "getValues": [Function],
            "initialValues": undefined,
            "initialize": [Function],
            "initialized": false,
            "invalid": false,
            "keepDirtyOnReinitialize": false,
            "persistentSubmitErrors": false,
            "pristine": true,
            "pure": true,
            "register": [Function],
            "registerField": [Function],
            "registerInnerOnSubmit": [Function],
            "registeredFields": undefined,
            "reset": [Function],
            "resetSection": [Function],
            "sectionPrefix": undefined,
            "setSubmitFailed": [Function],
            "setSubmitSucceeded": [Function],
            "shouldAsyncValidate": [Function],
            "shouldError": [Function],
            "shouldValidate": [Function],
            "shouldWarn": [Function],
            "startAsyncValidation": [Function],
            "startSubmit": [Function],
            "stopAsyncValidation": [Function],
            "stopSubmit": [Function],
            "submit": [Function],
            "submitFailed": false,
            "submitSucceeded": false,
            "submitting": false,
            "syncErrors": Object {},
            "syncWarnings": Object {},
            "touch": [Function],
            "touchOnBlur": true,
            "touchOnChange": false,
            "triggerSubmit": undefined,
            "unregister": [Function],
            "unregisterField": [Function],
            "untouch": [Function],
            "updateSyncErrors": [Function],
            "updateSyncWarnings": [Function],
            "updateUnregisteredFields": false,
            "valid": true,
            "validExceptSubmit": true,
            "validate": [Function],
            "values": Object {},
            "warning": undefined,
          }
        }
        className="form-control"
        component={[Function]}
        label="Email"
        name="email"
        normalize={[Function]}
      >
        <ConnectedField
          _reduxForm={
            Object {
              "anyTouched": false,
              "array": Object {
                "insert": [Function],
                "move": [Function],
                "pop": [Function],
                "push": [Function],
                "remove": [Function],
                "removeAll": [Function],
                "shift": [Function],
                "splice": [Function],
                "swap": [Function],
                "unshift": [Function],
              },
              "arrayInsert": [Function],
              "arrayMove": [Function],
              "arrayPop": [Function],
              "arrayPush": [Function],
              "arrayRemove": [Function],
              "arrayRemoveAll": [Function],
              "arrayShift": [Function],
              "arraySplice": [Function],
              "arraySwap": [Function],
              "arrayUnshift": [Function],
              "asyncErrors": undefined,
              "asyncValidate": [Function],
              "asyncValidating": false,
              "autofill": [Function],
              "blur": [Function],
              "change": [Function],
              "clearAsyncError": [Function],
              "clearFields": [Function],
              "clearSubmit": [Function],
              "clearSubmitErrors": [Function],
              "destroy": [Function],
              "destroyOnUnmount": true,
              "dirty": false,
              "dispatch": [Function],
              "enableReinitialize": false,
              "error": undefined,
              "fields": Array [
                "email",
                "password",
              ],
              "focus": [Function],
              "forceUnregisterOnUnmount": false,
              "form": "contact",
              "getFormState": [Function],
              "getValues": [Function],
              "initialValues": undefined,
              "initialize": [Function],
              "initialized": false,
              "invalid": false,
              "keepDirtyOnReinitialize": false,
              "persistentSubmitErrors": false,
              "pristine": true,
              "pure": true,
              "register": [Function],
              "registerField": [Function],
              "registerInnerOnSubmit": [Function],
              "registeredFields": undefined,
              "reset": [Function],
              "resetSection": [Function],
              "sectionPrefix": undefined,
              "setSubmitFailed": [Function],
              "setSubmitSucceeded": [Function],
              "shouldAsyncValidate": [Function],
              "shouldError": [Function],
              "shouldValidate": [Function],
              "shouldWarn": [Function],
              "startAsyncValidation": [Function],
              "startSubmit": [Function],
              "stopAsyncValidation": [Function],
              "stopSubmit": [Function],
              "submit": [Function],
              "submitFailed": false,
              "submitSucceeded": false,
              "submitting": false,
              "syncErrors": Object {},
              "syncWarnings": Object {},
              "touch": [Function],
              "touchOnBlur": true,
              "touchOnChange": false,
              "triggerSubmit": undefined,
              "unregister": [Function],
              "unregisterField": [Function],
              "untouch": [Function],
              "updateSyncErrors": [Function],
              "updateSyncWarnings": [Function],
              "updateUnregisteredFields": false,
              "valid": true,
              "validExceptSubmit": true,
              "validate": [Function],
              "values": Object {},
              "warning": undefined,
            }
          }
          asyncValidating={false}
          className="form-control"
          component={[Function]}
          dirty={false}
          dispatch={[Function]}
          label="Email"
          name="email"
          normalize={[Function]}
          pristine={true}
          syncError="Required"
        >
          <renderField
            className="form-control"
            input={
              Object {
                "name": "email",
                "onBlur": [Function],
                "onChange": [Function],
                "onDragStart": [Function],
                "onDrop": [Function],
                "onFocus": [Function],
                "value": "",
              }
            }
            label="Email"
            meta={
              Object {
                "active": false,
                "asyncValidating": false,
                "autofilled": false,
                "dirty": false,
                "dispatch": [Function],
                "error": "Required",
                "form": "contact",
                "initial": undefined,
                "invalid": true,
                "pristine": true,
                "submitFailed": false,
                "submitting": false,
                "touched": false,
                "valid": false,
                "visited": false,
                "warning": undefined,
              }
            }
          >
            <div>
              <label
                className="control-label"
              >
                Email
              </label>
              <div>
                <input
                  className="form-control"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Email"
                  value=""
                />
              </div>
            </div>
          </renderField>
        </ConnectedField>
      </Connect(ConnectedField)>
    </Field>
  </div>
  <div
    className="form-group"
    id="password"
  >
    <Field
      className="form-control"
      component={[Function]}
      label="Password"
      name="password"
      type="password"
    >
      <Connect(ConnectedField)
        _reduxForm={
          Object {
            "anyTouched": false,
            "array": Object {
              "insert": [Function],
              "move": [Function],
              "pop": [Function],
              "push": [Function],
              "remove": [Function],
              "removeAll": [Function],
              "shift": [Function],
              "splice": [Function],
              "swap": [Function],
              "unshift": [Function],
            },
            "arrayInsert": [Function],
            "arrayMove": [Function],
            "arrayPop": [Function],
            "arrayPush": [Function],
            "arrayRemove": [Function],
            "arrayRemoveAll": [Function],
            "arrayShift": [Function],
            "arraySplice": [Function],
            "arraySwap": [Function],
            "arrayUnshift": [Function],
            "asyncErrors": undefined,
            "asyncValidate": [Function],
            "asyncValidating": false,
            "autofill": [Function],
            "blur": [Function],
            "change": [Function],
            "clearAsyncError": [Function],
            "clearFields": [Function],
            "clearSubmit": [Function],
            "clearSubmitErrors": [Function],
            "destroy": [Function],
            "destroyOnUnmount": true,
            "dirty": false,
            "dispatch": [Function],
            "enableReinitialize": false,
            "error": undefined,
            "fields": Array [
              "email",
              "password",
            ],
            "focus": [Function],
            "forceUnregisterOnUnmount": false,
            "form": "contact",
            "getFormState": [Function],
            "getValues": [Function],
            "initialValues": undefined,
            "initialize": [Function],
            "initialized": false,
            "invalid": false,
            "keepDirtyOnReinitialize": false,
            "persistentSubmitErrors": false,
            "pristine": true,
            "pure": true,
            "register": [Function],
            "registerField": [Function],
            "registerInnerOnSubmit": [Function],
            "registeredFields": undefined,
            "reset": [Function],
            "resetSection": [Function],
            "sectionPrefix": undefined,
            "setSubmitFailed": [Function],
            "setSubmitSucceeded": [Function],
            "shouldAsyncValidate": [Function],
            "shouldError": [Function],
            "shouldValidate": [Function],
            "shouldWarn": [Function],
            "startAsyncValidation": [Function],
            "startSubmit": [Function],
            "stopAsyncValidation": [Function],
            "stopSubmit": [Function],
            "submit": [Function],
            "submitFailed": false,
            "submitSucceeded": false,
            "submitting": false,
            "syncErrors": Object {},
            "syncWarnings": Object {},
            "touch": [Function],
            "touchOnBlur": true,
            "touchOnChange": false,
            "triggerSubmit": undefined,
            "unregister": [Function],
            "unregisterField": [Function],
            "untouch": [Function],
            "updateSyncErrors": [Function],
            "updateSyncWarnings": [Function],
            "updateUnregisteredFields": false,
            "valid": true,
            "validExceptSubmit": true,
            "validate": [Function],
            "values": Object {},
            "warning": undefined,
          }
        }
        className="form-control"
        component={[Function]}
        label="Password"
        name="password"
        normalize={[Function]}
        type="password"
      >
        <ConnectedField
          _reduxForm={
            Object {
              "anyTouched": false,
              "array": Object {
                "insert": [Function],
                "move": [Function],
                "pop": [Function],
                "push": [Function],
                "remove": [Function],
                "removeAll": [Function],
                "shift": [Function],
                "splice": [Function],
                "swap": [Function],
                "unshift": [Function],
              },
              "arrayInsert": [Function],
              "arrayMove": [Function],
              "arrayPop": [Function],
              "arrayPush": [Function],
              "arrayRemove": [Function],
              "arrayRemoveAll": [Function],
              "arrayShift": [Function],
              "arraySplice": [Function],
              "arraySwap": [Function],
              "arrayUnshift": [Function],
              "asyncErrors": undefined,
              "asyncValidate": [Function],
              "asyncValidating": false,
              "autofill": [Function],
              "blur": [Function],
              "change": [Function],
              "clearAsyncError": [Function],
              "clearFields": [Function],
              "clearSubmit": [Function],
              "clearSubmitErrors": [Function],
              "destroy": [Function],
              "destroyOnUnmount": true,
              "dirty": false,
              "dispatch": [Function],
              "enableReinitialize": false,
              "error": undefined,
              "fields": Array [
                "email",
                "password",
              ],
              "focus": [Function],
              "forceUnregisterOnUnmount": false,
              "form": "contact",
              "getFormState": [Function],
              "getValues": [Function],
              "initialValues": undefined,
              "initialize": [Function],
              "initialized": false,
              "invalid": false,
              "keepDirtyOnReinitialize": false,
              "persistentSubmitErrors": false,
              "pristine": true,
              "pure": true,
              "register": [Function],
              "registerField": [Function],
              "registerInnerOnSubmit": [Function],
              "registeredFields": undefined,
              "reset": [Function],
              "resetSection": [Function],
              "sectionPrefix": undefined,
              "setSubmitFailed": [Function],
              "setSubmitSucceeded": [Function],
              "shouldAsyncValidate": [Function],
              "shouldError": [Function],
              "shouldValidate": [Function],
              "shouldWarn": [Function],
              "startAsyncValidation": [Function],
              "startSubmit": [Function],
              "stopAsyncValidation": [Function],
              "stopSubmit": [Function],
              "submit": [Function],
              "submitFailed": false,
              "submitSucceeded": false,
              "submitting": false,
              "syncErrors": Object {},
              "syncWarnings": Object {},
              "touch": [Function],
              "touchOnBlur": true,
              "touchOnChange": false,
              "triggerSubmit": undefined,
              "unregister": [Function],
              "unregisterField": [Function],
              "untouch": [Function],
              "updateSyncErrors": [Function],
              "updateSyncWarnings": [Function],
              "updateUnregisteredFields": false,
              "valid": true,
              "validExceptSubmit": true,
              "validate": [Function],
              "values": Object {},
              "warning": undefined,
            }
          }
          asyncValidating={false}
          className="form-control"
          component={[Function]}
          dirty={false}
          dispatch={[Function]}
          label="Password"
          name="password"
          normalize={[Function]}
          pristine={true}
          syncError="Required"
          type="password"
        >
          <renderField
            className="form-control"
            input={
              Object {
                "name": "password",
                "onBlur": [Function],
                "onChange": [Function],
                "onDragStart": [Function],
                "onDrop": [Function],
                "onFocus": [Function],
                "value": "",
              }
            }
            label="Password"
            meta={
              Object {
                "active": false,
                "asyncValidating": false,
                "autofilled": false,
                "dirty": false,
                "dispatch": [Function],
                "error": "Required",
                "form": "contact",
                "initial": undefined,
                "invalid": true,
                "pristine": true,
                "submitFailed": false,
                "submitting": false,
                "touched": false,
                "valid": false,
                "visited": false,
                "warning": undefined,
              }
            }
            type="password"
          >
            <div>
              <label
                className="control-label"
              >
                Password
              </label>
              <div>
                <input
                  className="form-control"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </renderField>
        </ConnectedField>
      </Connect(ConnectedField)>
    </Field>
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn singin-button"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
